specification/dartLangSpec.tex

```diff
@@ -6124,201 +6131,290 @@ \subsection{Constants}
+  It is further constant if $e_1$ is a constant expression and either
+  \begin{enumerate}
+  \item $e_1$ evaluates to \TRUE{}, or
+  \item $e_1$ evaluates to \FALSE{} and $e_2$ is a constant expression
+    that evaluates to an instance of type \code{bool}.
+  \end{enumerate}
 
 \item
   An expression of the form \code{\gtilde{}$e_1$} is
   a potentially constant expression
   if $e_1$ is a potentially constant expression.
   It is further a constant expression if $e_1$ is
-  a constant expression that evaluates to an instance of type \code{int}.
+  a constant expression that evaluates to an instance of type \code{int}
+  such that \gtilde{} denotes an instance operator invocation.
 
 \item An expression of one of the forms \code{$e_1$\,\&\,$e_2$},
   \code{$e_1$\,|\,$e_2$}, or \code{$e_1$\,\^\,$e_2$} is potentially constant
   if $e_1$ and $e_2$ are both potentially constant expressions.
   It is further constant if both $e_1$ and $e_2$ are constant expressions that
-  both evaluate to instances of \code{int}, or both to instances of \code{bool}.
-  % The bool case is new in 2.1.
+  both evaluate to instances of \code{int},
+  or both to instances of \code{bool},
+  such that the operator symbol
+  \lit{\&}, \lit{|}, respectively \lit{\^}
+  denotes an instance operator invocation.
 
 \item An expression of one of the forms \code{$e_1$\,\gtilde{}/\,$e_2$},
   \code{$e_1$\,\gtgt\,$e_2$}, \code{$e_1$\,\gtgtgt\,$e_2$},
   or \code{$e_1$\,\ltlt\,$e_2$} is potentially constant
   if $e_1$ and $e_2$ are both potentially constant expressions.
   It is further constant if both $e_1$ and $e_2$ are constant expressions that
-  both evaluate to an instance of \code{int}.
+  both evaluate to an instance of \code{int},
+  such that the operator symbol
+  \lit{\gtilde{}/}, \lit{\gtgt}, \lit{\gtgtgt}, respectively \lit{\ltlt}
+  denotes an instance operator invocation.
 
 \item An expression of the form \code{$e_1$\,+\,$e_2$} is
   a potentially constant expression if $e_1$ and $e_2$
   are both potentially constant expressions.
-  It is further a constant expression if both $e_1$ and $e_2$ are constant expressions
+  It is further a constant expression
+  if both $e_1$ and $e_2$ are constant expressions
   and either both evaluate to an instance of \code{int} or \code{double},
-  or both evaluate to an instance of \code{String}.
+  or both evaluate to an instance of \code{String},
+  such that \lit{+} denotes an instance operator invocation.
 
-\item An expression of the form \code{-$e_1$} is a potentially constant expression
+\item
+  An expression of the form \code{-$e_1$} is a potentially constant expression
   if $e_1$ is a potentially constant expression.
   It is further a constant expression if $e_1$ is a constant expression that
-  evaluates to an instance of type \code{int} or \code{double}.
+  evaluates to an instance of type \code{int} or \code{double},
+  such that \lit{-} invokes the instance operator \code{unary-}.
```

Comment:
You say "*the* instance operator" here, but "*an* instance operator" everywhere else.

Just say that `\lit{-} denotes an instance operator invocation.` like everywhere else. The other places don't say *which* one they denote.

Reply:
Done.